rmf_task/CMakeLists.txt

```
@@ -42,6 +51,8 @@ target_link_libraries(rmf_task
   PUBLIC
     rmf_battery::rmf_battery
     Threads::Threads
+    ${STD_CXX_FS}
+
```

I think you might be able to drop the conditional and variables and do something like this instead:
```suggestion
    $<$<CXX_COMPILER_ID:GNU>:stdc++fs>
```